### PR TITLE
daemonset editing actually updated pods REVERTED

### PR DESCRIPTION
**Pull request by mburke5678**
https://github.com/openshift/openshift-docs/issues/9623

REVERTED. See https://github.com/openshift/openshift-docs/pull/9897#issuecomment-406707318

### CODE REVIEW

**Comment by mburke5678** [comment]
@sjenning Can someone take a look at this change? Our docs were different compared to the Kubernetes docs regarding updating daemonsets and existing pods.

File: dev_guide/daemonsets.adoc [hunk 1]
```diff
@@ -121,8 +121,10 @@ Node:        openshift-node02.hostname.com/10.14.20.137
 
 [IMPORTANT]
 ====
-Currently, updating a daemonset's pod template does not affect existing pod
```
Comment (akostadinov) [inline dev_guide/daemonsets.adoc]: Maybe this line shouldn't have been removed, I haven't tested this.

Comment (sjenning) [reply dev_guide/daemonsets.adoc]: The existing document wasn't wrong, it was just incomplete in that if you change the nodeSelector in the pod template, that can cause pods to be assigned/removed from nodes with a potentially new template.  Nodes that still match the nodeSelector will not get the new pod template if it has been changed.

I would just add your clarifying sentence and not remove anything.

Comment (mburke5678) [reply dev_guide/daemonsets.adoc]: @sjenning PTAL

Comment (mburke5678) [reply dev_guide/daemonsets.adoc]: @mdshuai PTAL

File: dev_guide/daemonsets.adoc [hunk 2]
```diff
@@ -121,12 +121,15 @@ Node:        openshift-node02.hostname.com/10.14.20.137
...
+Updating a daemonset's pod template does not affect existing pod
 replicas. Moreover, if you delete a daemonset and then create a new daemonset
 with a different template but the same label selector, it will recognize any
 existing pod replicas as having matching labels and thus will not update them or
 create new replicas despite a mismatch in the pod template.
 
+If you change node labels, the DaemonSet adds pods to nodes matching the new labels and delete pods 
+from nodes that do not match the new lablels.
```
Comment (sjenning) [inline dev_guide/daemonsets.adoc]: typo: labels

File: dev_guide/daemonsets.adoc [hunk 3]
```diff
@@ -121,12 +121,15 @@ Node:        openshift-node02.hostname.com/10.14.20.137
...
 [IMPORTANT]
 ====
-Currently, updating a daemonset's pod template does not affect existing pod
+Updating a daemonset's pod template does not affect existing pod
 replicas. Moreover, if you delete a daemonset and then create a new daemonset
 with a different template but the same label selector, it will recognize any
 existing pod replicas as having matching labels and thus will not update them or
 create new replicas despite a mismatch in the pod template.
 
+If you change node labels, the DaemonSet adds pods to nodes matching the new labels and delete pods
```
Comment (sjenning) [inline dev_guide/daemonsets.adoc]: s/delete/deletes/

**Comment by mburke5678** [comment]
@mdshuai PTAL

**Comment by mburke5678** [comment]
@mdshuai BUMP ^^

**Comment by mdshuai** [comment]
This change lgtm, thanks for the update.

**Comment by mburke5678** [comment]
@sjenning @akostadinov @mdshuai 
Which versions should this change be applied? Aleksander positively tested in 3.9 and 3.10. Any other versions?

**Comment by mburke5678** [comment]
@openshift/team-documentation PTAL

File: dev_guide/daemonsets.adoc [hunk 4]
```diff
@@ -13,28 +13,28 @@ toc::[]
 
 == Overview
 
-A daemonset can be used to run replicas of a pod on specific or all nodes in an
+A DaemonSet can be used to run replicas of a pod on specific or all nodes in an
```
Comment (kalexand-rh) [inline dev_guide/daemonsets.adoc]: "You can use..."

File: dev_guide/daemonsets.adoc [hunk 5]
```diff
@@ -121,12 +121,17 @@ Node:        openshift-node02.hostname.com/10.14.20.137
 
 [IMPORTANT]
 ====
-Currently, updating a daemonset's pod template does not affect existing pod
-replicas. Moreover, if you delete a daemonset and then create a new daemonset
+* If you update a DaemonSet's pod template, the existing pod
+replicas are not affected. 
+
+* If you delete a DaemonSet and then create a new DaemonSet
 with a different template but the same label selector, it will recognize any
```
Comment (kalexand-rh) [inline dev_guide/daemonsets.adoc]: s/will recognize/recognizes

File: dev_guide/daemonsets.adoc [hunk 6]
```diff
@@ -121,12 +121,17 @@ Node:        openshift-node02.hostname.com/10.14.20.137
...
 existing pod replicas as having matching labels and thus will not update them or
```
Comment (kalexand-rh) [inline dev_guide/daemonsets.adoc]: s/will/does

File: dev_guide/daemonsets.adoc [hunk 7]
```diff
@@ -121,12 +121,17 @@ Node:        openshift-node02.hostname.com/10.14.20.137
...
 create new replicas despite a mismatch in the pod template.
 
-To update a daemonset, force new pod replicas to be created by deleting the old
+* If you change node labels, the DaemonSet adds pods to nodes matching the new labels and deletes pods
```
Comment (kalexand-rh) [inline dev_guide/daemonsets.adoc]: s/matching/that matches

File: dev_guide/daemonsets.adoc [hunk 8]
```diff
@@ -1,5 +1,5 @@
 [[dev-guide-daemonsets]]
-= Using Daemonsets
+= Using DaemonSets
```
Comment (kalexand-rh) [inline dev_guide/daemonsets.adoc]: Please update the title in the _topic_map.yaml file.

I see a couple of other uses of "Daemonsets" in the collection. I'd consider fixing them on this PR or a follow-up PR.

**Review by kalexand-rh — COMMENTED** [review]
Nits, because I can't help it, a request and a suggestion. The changes themselves look good!

File: dev_guide/daemonsets.adoc (hunk 8 above)
Comment (mburke5678) [reply dev_guide/daemonsets.adoc]: @kalexand-rh Updated instances of Daemonset and daemonset to DaemonSet.

File: dev_guide/daemonsets.adoc (hunk 7 above)
Comment (kalexand-rh) [reply dev_guide/daemonsets.adoc]: Yes! My apologies!

**Comment by mburke5678** [comment]
@aveshagarwal Can you help determine which versions this change applies to? @akostadinov tested in 3.9 and 3.10.

**Comment by aveshagarwal** [comment]
> @aveshagarwal Can you help determine which versions this change applies to? @akostadinov tested in 3.9 and 3.10.

@mburke5678 I am not sure what behavior change to DS, we are talking about, but the behavior described in this PR has existed as far as I remember.

**Comment by mburke5678** [comment]
@aveshagarwal 
I changed: _Currently, updating a daemonset’s pod template does not affect existing pod replicas._
To: _If node labels are changed, the DaemonSet adds pods to nodes matching the new labels and delete pods from nodes that do not match the new labels._ 
In https://docs.openshift.com/container-platform/3.9/dev_guide/daemonsets.html

Have daemonsets always deleted pods without matching labels after a change?

**Comment by aveshagarwal** [comment]
> Have daemonsets always deleted pods without matching labels after a change?

Yes, otherwise that would be an error to continue keeping pods on non-matching nodes.

**Comment by mburke5678** [comment]
In PR openshift/openshift-docs#9897, we tried to make the capitalization of DaemonSet consistent across all docs. But, the PR created too many merge errors, especially in the earlier branches (incl. files added that would need to be removed). I am reverting this PR. We can pursue the capitalization changes post-3.10.

**Pull request merged**